### PR TITLE
Removing unused build options and deprecated proof compile flag

### PR DESCRIPTION
**Pull request by HanielB**

### CODE REVIEW

**Comment by HanielB** [comment]
As things are, CI will always fail because the `production-dbg` build uses the `--lfsc` configure flag.

File: test/regress/run_regression.py [hunk 1]
```diff
@@ -12,8 +12,7 @@
 """
 Usage:
 
-    run_regression.py [--enable-proof] [--with-lfsc] [--dump]
-        [--use-skip-return-code] [wrapper] cvc4-binary
+    run_regression.py [--dump] [--use-skip-return-code] [wrapper] cvc4-binary
         [benchmark.cvc | benchmark.smt | benchmark.smt2 | benchmark.p]
```
Comment (mpreiner) [inline test/regress/run_regression.py:16]: This is outdated and doesn't reflect the current set of options. Maybe remove?

**Review by mpreiner — APPROVED** [review]
LGTM except for one minor comment.

**Comment by mpreiner** [comment]
> As things are, CI will always fail because the `production-dbg` build uses the `--lfsc` configure flag.

Why not remove the `--lfsc` from CI then? :-)

File: test/unit/api/solver_black.cpp [hunk 2]
```diff
@@ -1423,7 +1423,7 @@ TEST_F(TestApiBlackSolver, getUnsatCore3)
   {
     d_solver.assertFormula(t);
   }
-  Result res = d_solver.checkSat();
+  CVC4::api::Result res = d_solver.checkSat();
```
Comment (mpreiner) [inline test/unit/api/solver_black.cpp:1426]: Is this change really necessary? There is a `using namespace api` at the top of the file.

Comment (HanielB) [reply test/unit/api/solver_black.cpp:1426]: I copied what was going on in the other files. There is some ambiguity on which `Result` to use because of the namespaces.
